addons/survey/models/survey_survey.py

```diff
@@ -181,6 +185,42 @@ def _read_group_stage_ids(self, stages, domain, order):
         stage_ids = stages._search([], order=order, access_rights_uid=SUPERUSER_ID)
         return stages.browse(stage_ids)
 
+    @api.model
+    def create(self, vals):
+        if 'certification_badge_id' in vals:
+            survey = super(Survey, self).create(vals)
+            goal = self.env['gamification.goal.definition'].create({
+                'name': survey.title,
+                'description': "%s certification passed" % survey.title,
+                'domain': "['&', ('survey_id', '=', %s), ('quizz_passed', '=', True)]" % survey.id,
+                'computation_mode': 'count',
+                'display_mode': 'boolean',
+                'model_id': self.env.ref('survey.model_survey_user_input').id,
+                'condition': 'higher',
+                'batch_mode': True,
+                'batch_distinctive_field': self.env.ref('survey.field_survey_user_input__partner_id').id,
+                'batch_user_expression': 'user.partner_id.id'
+            })
+            if not survey.certification_badge_id.challenge_ids:
```

Comment:
Not sure here. Either we have a generic challenge for the reward (Certification Challenge), either one by survey. Not sure what is better, probably one generic challenge as it is basically linked to the reward, not really the survey itself.

Reply:
Yes, I also think the challenge can be generic for the reward (but we have to create anyway one challenge per badge.)
And if we finally conclude that badge is unique for each survey.. it doesn't change anything at the end.
Let's the fight begin ! :smiling_imp: :fist_right:  :fist_left: